clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -303,6 +303,26 @@ public void pollNoWakeup() {
         poll(time.timer(0), null, true);
     }
 
+    /**
+     * Poll for network IO in best-effort only; i.e. do not check any pending requests or metadata errors
+     * so that no exception should ever be thrown, also wakeups be triggered and no interrupted exception either.
+     */
+    public void pollNoThrow() {
```

Comment:
I'm not married to the name or approach here, just to illustrate what should be trimmed off from the `poll` call to make sure we never throw.

Reply:
I can't really think of a better name. Our intent here is just to send any requests which can be sent now, so I was considering a name like `transmitSends()`. What do you think?

Reply:
`transmitSends()` sounds good to me, will use that name and also replace other places when plausible with this function.

Reply:
@hachikuji I tried to use `transmitSends` in maybeLeaveGroup as well but there’s an testing issue with it: in many unit tests we check the returned future (although in non-testing code we never check the returned future) and that requires we always handle the response and set the future as well. pollNoWake calls firePendingCompletedRequests at the beginning and the end of the call so it guarantees that the future is completed under normal processing, but transmitSends does not, as a result it breaks a bunch of unit tests that checks the returned future..

So I think to keep this PR scope small I'd just change this part only but not others, WDYT?

Reply:
Sounds fine to me